From 95692289f165b5b11e196e233df1734ddc3ee874 Mon Sep 17 00:00:00 2001
From: Varun Sharma <vsharma@habana.ai>
Date: Thu, 30 Jun 2022 10:46:24 +0300
Subject: [PATCH] [SW-87926] ppboot: reset arc0 just before entering preboot

This patch brings arc0 out of reset so that preboot while
reading bootinfo in binning driver can access arc0 iccm.
In gaudi2 preboot, binning driver comes as part of zephyr os
and at that time for custom binning, we need to check if boot info entry
is available and apply masks. To access it arc0 should be out of reset
before accessing it, currently it was done quite late in flow.
Without it during reset like fw reset, preboot tries to read
boot info and goes into indefinite wait.

Change-Id: I141e8cefb7d734a5321c402606c1a6730b188ca5
Signed-off-by: Varun Sharma <vsharma@habana.ai>
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index bba8d2e..c04bec4 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -227,6 +227,8 @@
 	/* need to configure arb_timeout val before PLL is locked */
 	set_slv_cfg_reg_val();
 
+	preboot_cond_arc0_reset();
+
 	rst_src = hl_rst_src_get();
 	if (rst_src != HL_COLD_RST)
 		return;
